### PR TITLE
[Bugfix:InstructorUI] Fix empty teams being invisible

### PR DESCRIPTION
**Pull request by zhehua-zhang**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [x] Documentation has been updated/added if relevant
* [x] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
*partial* work for issue #9985. 
Currently, abandoned/empty teams in team gradeables are inaccessible and invisible in the gradeable details view, even if they have associated submissions. While there is existing code for an Empty Teams section, it does not work.

### What is the new behavior?
This PR partially fixes the empty teams section; empty teams are now visible and editing teams to re-add a student works as expected. The fix converts a `JOIN` statement that drops empty teams to a `LEFT JOIN` that preserves them.
image

### Known issues
Custom team names do not appear in the Empty Teams section, nor does a button to view submissions. Team history is broken for empty teams as well. However, prof Cutler requested that I split these issues into a separate PR.

### CODE REVIEW

**Comment by williamjallen** [comment]
It looks like there's a spelling error in your PR title.  The title is also pretty generic, and could be improved by succinctly describing what specifically is being fixed.  On a related note, I'm having trouble understanding what this PR is supposed to fix based on the PR description alone.  Can you please update the PR description, and add before and after screenshots?  Remember that your PR title and description are the permanent record of what and why a change was made, and someone may refer to this at a future point.

File: site/app/controllers/grading/ElectronicGraderController.php [hunk 1]
```diff
@@ -959,23 +959,26 @@ public function showDetails(string $gradeable_id) {
...
             foreach ($all_teams as $team) {
                 $student_ids = array_diff($student_ids, $team->getMembers());
                 $team_section = $gradeable->isGradeByRegistration() ? $team->getRegistrationSection() : $team->getRotatingSection();
                 if ($team->getSize() > 0 && (in_array($team_section, $sections) || $show_all)) {
                     $student_ids[] = $team->getLeaderId();
                 }
-                if ($team->getSize() === 0 && $show_empty_teams) {
-                    $empty_teams[] = $team;
-                }
+                //if ($team->getSize() === 0 && $show_empty_teams) {
```
Comment (ziesski) [inline site/app/controllers/grading/ElectronicGraderController.php:972]: Remove this for clarity

**Comment by ziesski** [comment]
should instructor have access to the submission of the empty team? Just from the screenshot, I could not tell because it does not have view button.

**Review by zacharymnp — APPROVED** [review]
Empty Teams dropdown exists for Instructor grading and not TA grading, which matches expected behavior. Empty Teams dropdown exists when a team is emptied and disappears when the team is filled again.

**Comment by ziesski** [comment]
Can you connect the newly created issues to this PR so future devs can use it as a reference?

**Review by ziesski — CHANGES_REQUESTED** [review]
Description mentions that this PR solves only some part of empty teams, however, it is not tackling the primary issue requested in issue #9985 that grading stats page does not correctly reflect abandoned teams. 

Here is an example, if you see section 1, there is only one valid team submission after removing one team. However, if we look at TA Grading By Sections: Section 1: 2 / 3 (66.7%), it is still 2/3 which is not correctly updated. 
Empty team is well displayed on the grading index page, matching the description, however, it also needs to match the grading stats page. Please fix them 

스크린샷 2024-06-04 오후 3 11 44

스크린샷 2024-06-04 오후 3 12 06

**Review by bmcutler — APPROVED** [review]
I tested this, it solves a significant part of the problem.  Thank you!
I will leave the issue open.  There are remaining bugs as noted above, and we need to write tests of empty/abandoned teams within the TA grading interface to make sure this feature doesn't break again in the future.
